### ISSUE TITLE
Need some clarification

### ISSUE THREAD

**Issue by kalai2033** [junyanz/pytorch-CycleGAN-and-pix2pix#996]
I am using both pix2pix and cycleGan models for my thesis research. I have a custom dataset with 417 image pairs. I have done a lots of experiments using your repositories. I have few questions. Some of my questions are generic as well. I use two gpus with 8GB memory each. so in total 16 GB GPU machine.

1) I understand the batch size as a hyperparamater. Currently i use 4 as a batch size for my experiments. But apart from increasing the training speed, what is the importance of batch size?
 
2) What does normalisation does? I use grayscale images during training. What is the difference between batch and instance normalization. And why it is not possible to use batch normalisation in multi GPU training?

3) Use of dropout and eval flags. When i train pix2pix i never use --dropout flag during testing but for cycleGAN i do. Why is it so? And when should i use --eval flag?

4) I use resnet9block generator as i train rectangular images. Is there any advantage of using unet architecture over resnet. I did not use unet since my images are 600*400.

5) I read a commit comment few days back about the semialigned dataset. Can you please brief about it and how to use it. Can it be used for both pix2pix and cycleGAN models?

6)Is it possible to save the training graphs from the visdom server?

@junyanz , Sorry for asking too many questions :).. Thanks for such a great repository.

**Comment by junyanz** [comment]
1. Batch Size is mostly for training speed especially you have multiple GPUs. 
2. Here is a [post](https://mlexplained.com/2018/11/30/an-overview-of-normalization-methods-in-deep-learning/) about normalization layers. For multi-GPUs, it is better to use synchronized batchnorm rather than the default batchnorm. See this [repo](https://github.com/vacancy/Synchronized-BatchNorm-PyTorch). 
3. We observe that `--dropout` may not help cyclegan as it contradicts the cycle-consistency loss. But you are free to try both. For eval, please see this [comment](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/test.py#L54). 
4. resnet_9blocks should be good enough. It is a more recent architecture compared to the unet.  
5. Not sure what is a semi-aligned dataset. 
6. You can save the plot from the visdom windows.

**Comment by kalai2033** [comment]
Thanks very much @junyanz .... I was talking about this pull request for semi aligned dataset [https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/pull/989]

**Comment by junyanz** [comment]
I see. The PR was from a different Github user and closed now. I haven't looked at the code. Maybe you can post your question under that PR and contact the author of that PR.
